### PR TITLE
News Updates Alerts

### LINKED ISSUE #2325: Stop displaying a "New" badge on the toolbar icon
Just realized that Test Pilot is trying to show a "New" badge on the icon. Too bad that basically no language will fit in a 4-characters space (the full word is "Novità", "Novi" is a non-sense in Italian, or a very good brand of chocolate…)

schermata 2017-03-16 alle 08 37 29

### LINKED ISSUE #2419: Add News alerts for major experiment changes
## How it works
- when we launch a *major* change or update to a feature (such as queues in Min Vid, or graduating page shot), the toolbar icon is badged (note, this feature goes beyond the `/experiments` news summary and is reserved for major updates.
- when the user clicks on the badge they are taken `/experiments` and a news modal is popped.
- if there are multiple new major features, the modal is a carousel
- if the updates have to do with specific experiments that the user is (a) not in and (b) eligible for (no conflicts installed), the option to enable said experiment is presented on the modal.

## Spec
Icon Badge: https://mozilla.invisionapp.com/share/GPBLHHLA6
Modal: https://mozilla.invisionapp.com/share/9YBLHHZKD

## Events
coming soon...

### LINKED ISSUE #2769: Improve handling of "new" badge on add-on toolbar button
Every [4 hours](https://github.com/mozilla/testpilot/blob/master/addon/src/webextension/lib/environments.js#L42), the add-on [checks for updates](https://github.com/mozilla/testpilot/blob/master/addon/src/webextension/lib/environments.js#L76) from the site. [If there are new or changed experiments](https://github.com/mozilla/testpilot/blob/master/addon/src/webextension/lib/browserAction.js#L35) since the last time the toolbar button was clicked, a "NEW" badge is shown.

This logic could be improved:

* "New" should clear immediately on click. Currently, [the timestamp is set on click](https://github.com/mozilla/testpilot/blob/master/addon/src/webextension/lib/browserAction.js#L39), but the badge isn't updated until the next data fetch from the site. That could be up to 4 hours away.

* "New" appears even if you've visited the site, seen the content, and installed all the new experiments (see also #2747). Maybe the "New" should be cleared on any visit to Test Pilot, regardless whether it was via clicking the toolbar button

### PR DESCRIPTION
**Pull request by meandavejustice**
# DO NOT MERGE UNTIL AFTER FREEZE ON THURSDAY

Opened in place of https://github.com/mozilla/testpilot/pull/2831 for storybook builds

- fixes #2419
- badge is now shown when news updates are available
- fixes #2325 fixes #2769 (replaced "new" with "!"
- added tests to HomePageWithAddon-test.js
- added js-cookie dependency
- bump addon version

Decided not to include a test file for the new modal component
since the functionality is duplicated in the tour modal. Opened

### CODE REVIEW

**Comment by lmorchard** [comment]
Hmm, looks like maybe the integration tests need to be updated to account for the news dialog that pops up? /cc @jrbenny35 - maybe this is simple? https://5853-40625180-gh.circle-artifacts.com/0/tmp/circle-artifacts.jmmGVPL/integration-test-results/ui-test-nightly.html

**Comment by meandavejustice** [comment]
@lmorchard just pushed at attempted fix, i now set a cookie to avoid showing the dialog

**Comment by lmorchard** [comment]
Heh, or maybe @meandavejustice already has this covered :) https://github.com/mozilla/testpilot/pull/2915/files#diff-2dc65831852a30ebd16fac68864075e1R27

**Comment by lmorchard** [comment]
# Storybook Deployment
Pull Request: http://testpilot-storybook.dev.mozaws.net/pull/2915/index.html
Commit: http://testpilot-storybook.dev.mozaws.net/9e1c758e3f4d8ad6cfedfdcc22072911bf6009cb/

**Comment by lmorchard** [comment]
# Storybook Deployment
Pull Request: http://testpilot-storybook.dev.mozaws.net/pull/2915/index.html
Commit: http://testpilot-storybook.dev.mozaws.net/73de8113e687ace11705c449d45e8a10cd0a2239/

**Comment by lmorchard** [comment]
# Storybook Deployment
Pull Request: http://testpilot-storybook.dev.mozaws.net/pull/2915/index.html
Commit: http://testpilot-storybook.dev.mozaws.net/bd42f935b2767ce46c6a1f482fd8e14952785107/

**Comment by lmorchard** [comment]
# Storybook Deployment
Pull Request: http://testpilot-storybook.dev.mozaws.net/pull/2915/index.html
Commit: http://testpilot-storybook.dev.mozaws.net/158df9656ebfbdf942de3140a4030864e6f944f4/

**Comment by lmorchard** [comment]
# Storybook Deployment
Pull Request: http://testpilot-storybook.dev.mozaws.net/pull/2915/index.html
Commit: http://testpilot-storybook.dev.mozaws.net/48f2d8a01279cb0dd5ea77f1c5fb5e45f689c31e/

**Comment by lmorchard** [comment]
# Storybook Deployment
Pull Request: http://testpilot-storybook.dev.mozaws.net/pull/2915/index.html
Commit: http://testpilot-storybook.dev.mozaws.net/4afbbc441b594a8de74ff1c2d990ed5c78af1c5b/

**Comment by lmorchard** [comment]
# Storybook Deployment
Pull Request: http://testpilot-storybook.dev.mozaws.net/pull/2915/index.html
Commit: http://testpilot-storybook.dev.mozaws.net/4023b140d34de183deef4734b41e5cea1c08e147/

**Comment by lmorchard** [comment]
# Storybook Deployment
Pull Request: http://testpilot-storybook.dev.mozaws.net/pull/2915/index.html
Commit: http://testpilot-storybook.dev.mozaws.net/4023b140d34de183deef4734b41e5cea1c08e147/

**Comment by lmorchard** [comment]
# Storybook Deployment
Pull Request: http://testpilot-storybook.dev.mozaws.net/pull/2915/index.html
Commit: http://testpilot-storybook.dev.mozaws.net/171bf0c1a49913aba753daad9952a3e1e647ab1c/

**Comment by lmorchard** [comment]
# Storybook Deployment
Pull Request: http://testpilot-storybook.dev.mozaws.net/pull/2915/index.html
Commit: http://testpilot-storybook.dev.mozaws.net/1ed46b17c158083311cc6df8e19b01d858f4b4a4/

**Comment by lmorchard** [comment]
# Storybook Deployment
Pull Request: http://testpilot-storybook.dev.mozaws.net/pull/2915/index.html
Commit: http://testpilot-storybook.dev.mozaws.net/7aac01d13e2791222855c936ecb0fd1d37c643e5/

**Comment by lmorchard** [comment]
# Storybook Deployment
Pull Request: http://testpilot-storybook.dev.mozaws.net/pull/2915/index.html
Commit: http://testpilot-storybook.dev.mozaws.net/0330bf17905f0ed68c14838ffeafd447f5fce414/

**Comment by meandavejustice** [comment]
Removed `isDev` flag and related test from news update selector. As our install integration test was failing because of this. I tried writing work arounds into the test, but because of the animation delay and since it only shows up sometimes depending of if there are updates with published dates in the future, it was really hard to get a consistent test

**Comment by lmorchard** [comment]
# Storybook Deployment
Pull Request: http://testpilot-storybook.dev.mozaws.net/pull/2915/index.html
Commit: http://testpilot-storybook.dev.mozaws.net/2332902f62533e95d1367bca5c2911f256547b61/

**Comment by lmorchard** [comment]
# Storybook Deployment
Pull Request: http://testpilot-storybook.dev.mozaws.net/pull/2915/index.html
Commit: http://testpilot-storybook.dev.mozaws.net/ec2ad8f265f6856f3943977409485801dda9dfc7/

**Comment by lmorchard** [comment]
# Storybook Deployment
Pull Request: http://testpilot-storybook.dev.mozaws.net/pull/2915/index.html
Commit: http://testpilot-storybook.dev.mozaws.net/ce91b672f2e08994ab69318097f4fd03bc1e5adc/

File: frontend/src/app/selectors/news.js [hunk 1]
```diff
@@ -6,16 +6,14 @@ import experimentSelector from './experiment';
 // from all available experiments and Test Pilot in general
 const newsUpdatesSelector = createSelector(
   store => store.news.updates,
-  store => store.browser.isDev,
```
Comment (lmorchard) [inline frontend/src/app/selectors/news.js:9]: Hmm, trying to remember what use `isDev` was in here... Maybe it's made obsolete by having the component states demoed in Storybook.

Comment (meandavejustice) [reply frontend/src/app/selectors/news.js:9]: I think it was just for showing news updates with published dates in the future

**Review by lmorchard — COMMENTED** [review]
Tests pass, things look good & seem to work well. Two new strings here, so I think we probably want to merge after the freeze on Thursday to catch the next deployment. 

[Alternatively, we could try re-using some existing strings.](https://github.com/mozilla/testpilot/pull/2831#discussion_r143034655) But, maybe that's not ideal?

**Comment by SoftVision-PaulOiegas** [comment]
@meandavejustice @lmorchard Is there a way we can force these functionalities to appear in order to test them whenever we ran our full stack of test on stage server? Otherwise, if something is broken now or gets broken in future, we won't be able to signal it until other news or major changes appear. And I imagine that won't be pretty to fix since it will be very short notice.

**Comment by lmorchard** [comment]
> Is there a way we can force these functionalities to appear in order to test them whenever we ran our full stack of test on stage server?

Not really a way to force them to appear. But, the dialog will reappear in the future when there are fresh news items or if you clear cookies that signal the dialog has been seen. 

There's also [the Storybook rendering](http://testpilot-storybook.dev.mozaws.net/e405f859139c58445daafc7f6809e19987e1c567/?knob-Has%20Addon=false&knob-Enabled=false&knob-Experiment=%7B%22title%22%3A%22Sample%20experiment%22%2C%22description%22%3A%22This%20is%20an%20example%20experiment%22%2C%22subtitle%22%3A%22%22%2C%22slug%22%3A%22snooze-tabs%22%2C%22survey_url%22%3A%22https%3A%2F%2Fexample.com%22%2C%22created%22%3A%222010-06-21T12%3A12%3A12Z%22%2C%22modified%22%3A%222010-06-21T12%3A12%3A12Z%22%7D&selectedKind=NewsUpdatesDialog&selectedStory=testpilot%20update&full=0&down=1&left=1&panelRight=0&downPanel=storybook%2Factions%2Factions-panel) that we can use to at least demo the components. That's not integrated into the site, but we can at least use that to walk through the states of the components without having to build the whole site with temporary experiment content

**Comment by SoftVision-PaulOiegas** [comment]
Uh, that's a bit sad. I would have been nice if we could at least make them resurface by changing the system date to one that had this kind of news added. That way we would have been always sure that the functionality is still in good health with all the changes that enter in between the news.

**Comment by lmorchard** [comment]
> make them resurface by changing the system date to one that had this kind of news added

That would probably work, as long as the cookie is deleted.

**Comment by SoftVision-PaulOiegas** [comment]
Nice, we'll try that tomorrow. Thanks!

**Pull request merged**